packages/components/src/components/stateful/ProductsTopAndBottom/ProductsTopAndBottom.tsx

```diff
@@ -55,29 +57,30 @@ function ProductsTopAndBottom(
     topSelectedRows,
     topProducts,
     setNbBottomRows,
+    setNbTopRows,
     sortValue,
     searchValue,
+    nbTopProducts,
   } = props
   const { t } = useTranslation('categories')
 
   const topProductsIds = topProducts
     .map((topProduct) => topProduct.productId)
     .sort()
 
-  const { current } = ref
+  const configuration = useAppSelector(selectConfiguration)
+
   useEffect(() => {
-    if (current && searchValue) {
-      current.scrollIntoView({ behavior: 'smooth' })
+    if (topProducts.length === 0) {
+      setNbTopRows(0)
     }
-  }, [current, searchValue])
-
-  const configuration = useAppSelector(selectConfiguration)
+  }, [topProducts, setNbTopRows])
```

Comment:
@Ponefar I don't understand why this is needed because you already set the number of top products in the  `useEffect` of `TopTable` component.
Can this be removed ?

Reply:
Je vais le faire en français ça sera plus simple :clown_face: , le seul moyen qu'on a de compter le nombre de top produit(s) est de calculer la longueur du tableau avec la position des top produits. Or lorsqu'on fait une recherche qui ne renvoie aucun résultat, si on suit le raisonnement alors on devrait avoir la longueur du tableau avec la position des top produits = 0. Cela signifie que si un utilisateur épingle un produit puis fait une recherche qui ne contient pas ce même produit, alors ses modifications seront annulées, c'est pour cela que j'ai différencier les deux. Si ce n'est pas clair on peut faire un call

Reply:
J'avoue que c'est pas très clair non